Tests/test_map.gd

```diff
@@ -12,3 +12,125 @@ func test_map_gen() -> void:
 	for couple: Array[int] in expected_exists_array:
 		var _room: RoomBase = test_map.rooms[couple[0]][couple[1]]
 		assert_not_null(_room, "Expected not null at %s but got %s" % [couple, _room])
+
+## Rule 0: Rooms should not have no event
+func test_rule_0_map_gen() -> void:
+	var expected_exists_array: Array[Array] = [[0,2],[1,1],[1,2],[1,3],[2,0],[2,1],[2,2],[2,3],[2,4],[3,1],[3,2],[3,3],[4,2]]
+
+	for couple: Array[int] in expected_exists_array:
+		var _room_event: EventBase = test_map.rooms[couple[0]][couple[1]].room_event
+		assert_not_null(_room_event, "Expected not null event at %s but got %s" % [couple, _room_event])
+
+## Rule 1: No Heal room before half of the map
+func test_rule_1_map_gen() -> void:
+	var current_room: RoomBase
+	for index_height: int in range(int(test_map.rooms.size()/2.0)):
```

Comment:
if you modify the `int` to a `ceil` / `floor` as I mentioned in the map generation, don't forget to change it here too